fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -3344,4 +3368,35 @@ public Bundle doHistory(MultivaluedMap<String, String> queryParameters, String r
 
         return bundleBuilder.build();
     }
-}
+
+    @Override
+    public ResourceEraseRecord doErase(FHIROperationContext operationContext, EraseDTO eraseDto) throws FHIROperationException {
+        // @implNote doReindex has a nice pattern to handle some retries in case of deadlock exceptions
+        final int TX_ATTEMPTS = 5;
+        int attempt = 1;
+        ResourceEraseRecord eraseRecord = new ResourceEraseRecord();
+        do {
+            FHIRTransactionHelper txn = null;
```

Comment:
I don't think it's worth changing right away, but the new transaction API is AutoCloseable so can use the try-with-resource pattern. Just need to setRollbackOnly if there's a failure. It eliminates excess boilerplate stuff, which tightens things up a bit.

Reply:
Hum - is there another TransactionHelper class? the current one is not autocloseable.

Reply:
Yeah, there's no need to use the helper anymore. It's just a wrapper which was kept for backwards compatibility. We may have an issue to clean this up throughout. If not, I'll create one for the backlog.

Reply:
Sorry...it didn't end up being AutoCloseable, but the pattern is: 
```
        // Must have a transaction in place because this done before the class test methods
        persistence.getTransaction().begin();
        try {
            saveBasicResource(getBasicResource());
            createCompositionReferencingSavedResource();
        } finally {
            persistence.getTransaction().end();
        }
```

The end() determines if it's necessary to commit or rollback.

Reply:
Ah... OK - I'll leave as-is for now, and we can cleanup in the followup issue (since I think it had a wider radius)